lib/puppet/settings.rb

```diff
@@ -1176,11 +1176,15 @@ def add_environment_resources(catalog, sections)
     if configured_environment == "production" && envdir && Puppet::FileSystem.exist?(envdir)
       configured_environment_path = File.join(envdir, configured_environment)
       if !Puppet::FileSystem.symlink?(configured_environment_path)
-        catalog.add_resource(
-          Puppet::Resource.new(:file,
-                               configured_environment_path,
-                               :parameters => { :ensure => 'directory' })
-        )
+        parameters = { :ensure => 'directory' }
+        unless Puppet::FileSystem.exist?(configured_environment_path)
+          parameters.merge!({ :mode => '0750' })
```

Comment:
Doesn't need to change, just noting it for future reference: You don't need to include the `{ }` in this case. Ruby will automatically make it a Hash that gets passed in as a single argument to `merge!()`. It behaves the same was as the last argument ("`:parameters => parameters`") being a Hash does with the `Puppet::Resource.new` does a few lines below here.